lib/src/git.rs

```diff
@@ -348,7 +366,38 @@ pub fn export_refs(
             failed_branches.push(branch_name);
         }
     }
-    Ok(failed_branches)
+    Ok((GitRefView::default(), failed_branches))
+}
+
+/// Reads the last seen state of git refs from a file, updates it with the
+/// provided function, and writes the result back to disk.
+///
+/// The state currently tracks only the local branches.
```

Comment:
This became a confusing question for me once I started thinking about the issue of concurrency and distributed filesystems, where we can't really trust git. Ignoring those questions, you are probably right.

However, if a repo is on Dropbox and the remote-tracking branch in git gets corrupted, the current behavior of creating a conflict is nice. Or should we just lose the remote tracking branch position in that case? This becomes more important when we try to do something like `--force-with-lease`.

I'm also not sure if the same answer is good for colocated and non-colocated repositories.

Reply:
I agree that this is confusing, but I don't see the contradiction between these two comments. I think both are saying that we should store our own concept of remote-tracking branches in the view object but probably not in the new file introduced by this PR.

>  if a repo is on Dropbox and the remote-tracking branch in git gets corrupted, the current behavior of creating a conflict is nice

Yes, I agree. Ah, it was probably my "Do you think we need to track remote-tracking branches too?" that sounded like I was saying that we shouldn't even track them in our model? I meant it in the context of the new file introduced by this PR, i.e. that I don't think we need to track remote-tracking branches in that file.

Makes sense?

Reply:
>  sounded like I was saying that we shouldn't even track them in our model?

Yes, that's what I was thinking about.

>  I don't think we need to track remote-tracking branches in that file. Makes sense?

I think it makes sense in practice; that's what the reality would be after this PR. I'm not 100% sure it works in theory :). In other words, it makes sense in the short term, and I'm still confused about the long term.

Mostly, that's because I was hoping that we could have a framework where exported branches and remote-tracking branches are treated in the same way. That may simply be a flawed idea, but I haven't adjusted my mind to a consistent theory of how it should work otherwise. I guess we forget jj's view of remote tracking branches whenever there is a conflict or git loses this information and wait for the next fetch. Hopefully this will work OK for our version of `--force-with-lease`.

Reply:
I guess what I'm saying in the last paragraph is that I'm not sure there's a behavior that keeps the remote-tracking branches in jj's model, makes sense, doesn't track the conflict base for conflits between jj and git, works with undo (or can we compromise here?), and generates benefit compared to only tracking them in git. I'm also not sure it doesn't exist. Finally, I'm also not sure how the current jj behavior measures up on the last two points.